Model/Method/PayPerEmail.php

```diff
@@ -206,6 +208,9 @@ private function getPayperemailService($payment)
         /** @var \Buckaroo\Magento2\Model\ConfigProvider\Method\PayPerEmail $config */
         $config = $this->configProviderMethodFactory->get('payperemail');
         $storeId = $payment->getOrder()->getStoreId();
+        /** @var \Buckaroo\Magento2\Model\ConfigProvider\Method\PayPerEmail $config */
+
+        $middleName = $this->_checkoutSession->getQuote()->getBillingAddress()->getMiddlename();
```

Comment:
To get the middleName is better to get it from order object from the payment, $payment->getOrder()->getBillingAddress()->getMiddlename(). Also CheckoutSession is resource-intensive class that should be instatiated on constructor, you should do it only as a Proxy.